src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -0,0 +1,278 @@
+        /// <param name="value">The value to be written as a JSON string as part of the name/value pair.</param>
+        /// <param name="suppressEscaping">If this is set, the writer assumes the property name is properly escaped and skips the escaping step.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified property name is too large.
+        /// </exception>
+        /// <exception cref="JsonWriterException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteString(ReadOnlySpan<byte> propertyName, DateTime value, bool suppressEscaping = false)
+        {
+            JsonWriterHelper.ValidateProperty(ref propertyName);
+
+            if (!suppressEscaping)
+                WriteStringSuppressFalse(ref propertyName, value);
+            else
+                WriteStringByOptions(ref propertyName, value);
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.String;
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<char> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<byte> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringEscapeProperty(ref ReadOnlySpan<char> propertyName, DateTime value, int firstEscapeIndexProp)
+        {
+            Debug.Assert(int.MaxValue / MaxExpansionFactorWhileEscaping >= propertyName.Length);
+
+            char[] propertyArray = null;
+
+            int length = firstEscapeIndexProp + MaxExpansionFactorWhileEscaping * (propertyName.Length - firstEscapeIndexProp);
+            Span<char> span;
+            if (length > StackallocThreshold)
+            {
+                propertyArray = ArrayPool<char>.Shared.Rent(length);
+                span = propertyArray;
+            }
+            else
+            {
+                // Cannot create a span directly since the span gets exposed outside this method.
```

Comment:
It's not returned, though.  Are you referring to a compiler error when you try to stackalloc into the Span directly?  If so, can you just define the span initially as:
```C#
Span<char> span = stackalloc char[0];
```
That shouldn't have any additional cost and should silence the error and let you make this:
```C#
span = stackalloc char[length];
```

Reply:
You can also just word it instead as:
```C#
Span<char> span = length <= StackallocThreshold ?
    stackalloc char[StackallocThreshold] : 
    (propertyArray = ArrayPool<char>.Shared.Rent(length));
```

Reply:
It doesn't look like I can:
> System\Text\Json\Writer\Utf8JsonWriter.cs(443,33): error CS8352: Cannot use local 'propertyNameSpan' in this context because it may expose referenced variables outside of their declaration scope [E:\GitHub\Fork\corefx\src\System.Text.Json\src\System.Text.Json.csproj]

Reply:
> It doesn't look like I can:

What if you stop passing propertyName by ref?